107/src/main/java/org/ehcache/jsr107/Eh107Configuration.java

```diff
@@ -78,8 +78,8 @@ private Eh107ConfigurationWrapper(CacheConfiguration<K, V> cacheConfiguration) {
 
     @Override
     public boolean isStoreByValue() {
-      OnHeapStoreServiceConfiguration heapConfig = findSingletonAmongst(OnHeapStoreServiceConfiguration.class, cacheConfiguration.getServiceConfigurations());
-      return heapConfig != null && heapConfig.storeByValue();
+      CopierConfiguration copierConfig = findSingletonAmongst(CopierConfiguration.class, cacheConfiguration.getServiceConfigurations());
+      return copierConfig == null;
```

Comment:
Condition is inverted here: It must be when there is no `copierConfig` then it is `storeByRef` and if there is one then it is `storeByValue`.